scipy/interpolate/_polyint.py

```diff
@@ -677,8 +743,91 @@ def _evaluate(self, x):
                 p[r[:-1]] = self.yi[r[-1]]
         return p
 
+    def derivative(self, x, der=1):
+        """
+        Evaluate one derivative of the polynomial at the point x
 
-def barycentric_interpolate(xi, yi, x, axis=0):
+        Parameters
+        ----------
+        x : array_like
+            Point or points at which to evaluate the derivatives
+        der : integer, optional
+            Which derivative to extract. This number includes the
+            function value as 0th derivative.
+
+        Returns
+        -------
+        d : ndarray
+            Derivative interpolated at the x-points. Shape of d is
+            determined by replacing the interpolation axis in the
+            original array with the shape of x.
+        """
+        x, x_shape = self._prepare_x(x)
+        y = self._evaluate_derivatives(x, der+1, all_lower=False)
+        return self._finish_y(y, x_shape)
+
+    def _evaluate_derivatives(self, x, der=None, all_lower=True):
+        # NB: der here is not the order of the highest derivative;
+        # instead, it is the size of the derivatives matrix that
+        # would be returned with all_lower=True, including the
+        # '0th' derivative (the undifferentiated function).
+        # E.g. to evaluate the 5th derivative alone, call
+        # _evaluate_derivatives(x, der=6, all_lower=False).
+
+        if (not all_lower) and (x.size == 0 or self.r == 0):
+            return np.zeros((0, self.r), dtype=self.dtype)
+
+        if (not all_lower) and der == 1:
+            return self._evaluate(x)
+
+        if der is None:
+            der = self.n
+
+        if all_lower and (x.size == 0 or self.r == 0):
+            cn = np.zeros((der, len(x), self.r), dtype=self.dtype)
+            return cn
+
+        if self._diff_cij is None:
+            # c[i,j] = xi[i] - xi[j]
```

Comment:
Can you remove the commented-out code

Reply:
These comments aren't code - they use matrix index notation to indicate the state of the `c` array after each successive in-place manipulation (which otherwise requires some thought to decipher and compare to the Berrut & Trefethen equations).